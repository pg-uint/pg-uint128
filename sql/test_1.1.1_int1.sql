-- Testing cross-types compatibility

SELECT (0::int1)::numeric;
SELECT ('0'::numeric)::int1;
SELECT ('128'::numeric)::int1;
SELECT (0::int1)::json;
SELECT json_build_object('some', 0::int1);
SELECT ('0'::json)::int1;
SELECT (0::int1)::jsonb;
SELECT jsonb_build_object('some', 0::int1);
SELECT ('0'::jsonb)::int1;
SELECT ('"abc"'::jsonb)::int1;
