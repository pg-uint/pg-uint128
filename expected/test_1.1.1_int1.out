-- Testing cross-types compatibility
SELECT (0::int1)::json;
 json 
------
 0
(1 row)

SELECT json_build_object('some', 0::int1);
 json_build_object 
-------------------
 {"some" : 0}
(1 row)

SELECT ('0'::json)::int1;
 int1 
------
 0
(1 row)

SELECT (0::int1)::jsonb;
 jsonb 
-------
 0
(1 row)

SELECT jsonb_build_object('some', 0::int1);
 jsonb_build_object 
--------------------
 {"some": 0}
(1 row)

SELECT ('0'::jsonb)::int1;
 int1 
------
 0
(1 row)

SELECT ('"abc"'::jsonb)::int1;
ERROR:  cannot cast jsonb string to type int1
